main/ry835ai.go

```diff
@@ -586,44 +618,131 @@ func processNMEALine(l string) (sentenceUsed bool) {
 			if err1 != nil {
 				return false
 			}
-			tmpSituation.Satellites = uint16(sat)
+			tmpSituation.Satellites = uint16(sat) // this seems to be reliable. UBX,03 handles >12 satellites solutions correctly.
 
 			// We've made it this far, so that means we've processed "everything" and can now make the change to mySituation.
 			mySituation = tmpSituation
 			return true
-		} else if x[1] == "03" { // satellite status message
+		} else if x[1] == "03" { // satellite status message. Only the first 20 satellites will be reported in this message for UBX firmware older than v3.0. Order seems to be GPS, then SBAS, then GLONASS.
 
 			// field 2 = number of satellites tracked
-			satSeen := 0 // satellites seen (signal present)
+			//satSeen := 0 // satellites seen (signal present)
 			satTracked, err := strconv.Atoi(x[2])
```

Comment:
Shouldn't rely on this, better to do as below based on the length of the actual message - in the case of a truncated message the below `for {}` block will cause an index overflow crash.